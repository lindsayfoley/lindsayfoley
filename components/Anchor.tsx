import React from "react";

export interface IAnchorProps {
  link: string;
  icon?: string;
  cta?: string;
  title?: string;
  external?: boolean;
}

const Anchor: React.FunctionComponent<IAnchorProps> = ({
  link,
  icon,
  cta,
  title,
  external
}) => {

  const anchorProps = {
    href: link,
    title
  }

  const props = external
    ? {
      target: "_blank",
      rel: "noreferrer",
      ...anchorProps
    } : {
      ...anchorProps
    }

  return (
    <a {...props}>
      {icon && <i className={icon}></i>}{cta}
    </a>
  )
}

export default Anchor;