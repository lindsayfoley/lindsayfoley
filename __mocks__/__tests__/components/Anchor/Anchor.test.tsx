import React from 'react';
import { shallow } from 'enzyme';
import Anchor, { IAnchorProps } from '../../../../components/Anchor';

const defaultProps: IAnchorProps = {
    link: '/portfolio',
    icon: 'fa-github-alt',
    cta: 'View my GitHub',
    title: 'Find me on GitHub;',
    external: false
}

const setup = (props = {}) => {
    const setupProps: IAnchorProps = { link: defaultProps.link, ...props }
    return shallow(<Anchor {...setupProps} />);
};

describe('Anchor', () => {

    it('renders correctly', () => {
        const wrapper = setup(defaultProps);
        expect(wrapper).toMatchSnapshot();
    });

    it('renders a link', () => {
        const wrapper = setup();
        expect(wrapper.find('[href]').length).toBe(1);
    });

    it('renders an icon', () => {
        const wrapper = setup({ icon: defaultProps.icon });
        expect(wrapper.find('i').length).toBe(1);
        expect(wrapper.find('i').hasClass(defaultProps.icon));
    });

    it('adds a title attribute', () => {
        const wrapper = setup({ title: defaultProps.title });
        expect(wrapper.find('a').props()).toHaveProperty('title', defaultProps.title);
    });

    it('displays a cta', () => {
        const wrapper = setup({ cta: defaultProps.cta });
        expect(wrapper.find('a').text()).toEqual(defaultProps.cta);
    });

    it('adds extra attributes to links when external is true', () => {
        const wrapper = setup({ external: true });
        expect(wrapper.find('a').prop('target')).toEqual('_blank');
        expect(wrapper.find('a').prop('rel')).toEqual('noreferrer');
    });
});