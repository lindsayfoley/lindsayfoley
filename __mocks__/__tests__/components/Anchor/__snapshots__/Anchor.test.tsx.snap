// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Anchor renders correctly 1`] = `
<a
  href="/portfolio"
  title="Find me on GitHub;"
>
  <i
    className="fa-github-alt"
  />
  View my GitHub
</a>
`;
